xrmocap/human_perception/keypoints_estimation/mmpose_trt_top_down_estimator.py

```diff
@@ -0,0 +1,555 @@
+                    Shape of the nested lists is
+                    (n_frame, n_human, n_keypoints, 3).
+                    Each keypoint is an array of (x, y, confidence).
+                heatmap_list (list):
+                    A list of keypoint heatmaps. len(heatmap_list) == n_frame
+                    and the shape of heatmap_list[f] is
+                    (n_human, n_keypoints, width, height).
+        """
+        image_array = video_to_array(input_path=video_path, logger=self.logger)
+        ret_kps_list, ret_heatmap_list, ret_boox_list = self.infer_array(
+            image_array=image_array,
+            bbox_list=bbox_list,
+            disable_tqdm=disable_tqdm,
+            return_heatmap=return_heatmap)
+        return ret_kps_list, ret_heatmap_list, ret_boox_list
+
+    def get_keypoints_from_result(
+            self, kps2d_list: List[list]) -> Union[Keypoints, None]:
+        """Convert returned keypoints2d into an instance of class Keypoints.
+
+        Args:
+            kps2d_list (List[list]):
+                A list of human keypoints, returned by
+                infer methods.
+                Shape of the nested lists is
+                (n_frame, n_human, n_keypoints, 3).
+
+        Returns:
+            Union[Keypoints, None]:
+                An instance of Keypoints with mask and
+                convention, data type is numpy.
+                If no one has been detected in any frame,
+                a None will be returned.
+        """
+        # shape: (n_frame, n_human, n_keypoints, 3)
+        n_frame = len(kps2d_list)
+        human_count_list = [len(human_list) for human_list in kps2d_list]
+        if len(human_count_list) > 0:
+            n_human = max(human_count_list)
+        else:
+            n_human = 0
+        n_keypoints = get_keypoint_num(self.get_keypoints_convention_name())
+        if n_human > 0:
+            kps2d_arr = np.zeros(shape=(n_frame, n_human, n_keypoints, 3))
+            mask_arr = np.ones_like(kps2d_arr[..., 0], dtype=np.uint8)
+            for f_idx in range(n_frame):
+                if len(kps2d_list[f_idx]) <= 0:
+                    mask_arr[f_idx, ...] = 0
+                    continue
+                for h_idx in range(n_human):
+                    if h_idx < len(kps2d_list[f_idx]):
+                        mask_arr[f_idx, h_idx, ...] = np.sign(
+                            np.array(kps2d_list[f_idx][h_idx])[:, -1])
+                        kps2d_arr[f_idx,
+                                  h_idx, :, :] = kps2d_list[f_idx][h_idx]
+                    else:
+                        mask_arr[f_idx, h_idx, ...] = 0
+            keypoints2d = Keypoints(
+                kps=kps2d_arr,
+                mask=mask_arr,
+                convention=self.get_keypoints_convention_name(),
+                logger=self.logger)
+        else:
+            keypoints2d = None
+        return keypoints2d
+
+    def inference_top_down_pose_model(self,
```

Comment:
Is there a chance import this function from mmdeploy?

Reply:
this function is rewrited to adapt the input of the original function `infer_array` of `mmpose_top_down_estimator`